File: backend/app/domain/entities/specialty.py
from dataclasses import dataclass


@dataclass(frozen=True)
class Specialty:
    """Catálogo de especialidades. "Ciclo Básico" es la única con
    is_basic_cycle=True — nunca la elige el usuario a mano, se asigna sola
    a cursos de 1ro a 3ro."""

    id: str
    name: str
    is_basic_cycle: bool


@dataclass(frozen=True)
class SchoolSettings:
    """Fila única de configuración general: techo de año de cursada y de
    división ofrecidos en los desplegables de alta de curso, más los
    umbrales que usa el panel de notificaciones (campana del header)."""

    max_grade_year: int
    max_division: int
    consecutive_absences_alert_threshold: int = 3
    preceptor_temp_assignment_alert_days: int = 2
    schedule_exception_alert_days: int = 2

File: frontend/lib/domain/entities/specialty.dart
/// Domain entity: Specialty
///
/// Catálogo de especialidades (Programación, Construcciones, Electrónica,
/// Ciclo Básico). "Ciclo Básico" es la única con `isBasicCycle == true` —
/// nunca se elige a mano, se asigna sola a cursos de 1ro a 3ro.
class Specialty {
  const Specialty({required this.id, required this.name, required this.isBasicCycle});

  final String id;
  final String name;
  final bool isBasicCycle;

  factory Specialty.fromJson(Map<String, dynamic> json) => Specialty(
        id: json['id'].toString(),
        name: json['name'],
        isBasicCycle: json['is_basic_cycle'] ?? false,
      );
}

/// Domain entity: SchoolSettings
///
/// Fila única de configuración general — techo de año de cursada y de
/// división que se ofrecen en los desplegables de alta de curso.
class SchoolSettings {
  const SchoolSettings({required this.maxGradeYear, required this.maxDivision});

  final int maxGradeYear;
  final int maxDivision;

  factory SchoolSettings.fromJson(Map<String, dynamic> json) => SchoolSettings(
        maxGradeYear: (json['max_grade_year'] as num).toInt(),
        maxDivision: (json['max_division'] as num).toInt(),
      );
}
